vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -0,0 +1,121 @@
+
+    # Convenience wrapper/syntactic sugar.
+    def __call__(self, **kwargs):
+        kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))
+        return self.figure(**kwargs)
+
+    # Convenience wrapper/syntactic sugar.
+    def __getitem__(self, arg_name: str):
+        # See table implementation for more details.
+        if arg_name == "type":
+            return self.type
+        return self.figure[arg_name]
+
+    # Interaction methods
+    @property
+    def _filter_interaction_input(self):
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "cellClicked": State(component_id=self._callable_object_id, component_property="cellClicked"),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
+
+    def _filter_interaction(
+        self, data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+    ) -> pd.DataFrame:
+        """Function to be carried out for pre-defined `filter_interaction`."""
+        ctd_cellClicked = ctd_filter_interaction["cellClicked"]
+        if not ctd_cellClicked["value"]:
+            return data_frame
+
+        # ctd_active_cell["id"] represents the underlying table id, so we need to fetch its parent Vizro Table actions.
+        source_table_actions = _get_component_actions(_get_parent_vizro_model(ctd_cellClicked["id"]))
+
+        for action in source_table_actions:
+            if action.function._function.__name__ != "filter_interaction" or target not in action.function["targets"]:
+                continue
+            column = ctd_cellClicked["value"]["colId"]
+            clicked_data = ctd_cellClicked["value"]["value"]
+            data_frame = data_frame[data_frame[column].isin([clicked_data])]
+
+        return data_frame
+
+    @_log_call
+    def pre_build(self):
+        if self.actions:
+            kwargs = self.figure._arguments.copy()
+
+            # taken from table implementation - see there for details
+            kwargs["data_frame"] = pd.DataFrame()
+            underlying_aggrid_object = self.figure._function(**kwargs)
+
+            if not hasattr(underlying_aggrid_object, "id"):
+                raise ValueError(
+                    "Underlying `AgGrid` callable has no attribute 'id'. To enable actions triggered by the `AgGrid`"
+                    " a valid 'id' has to be provided to the `AgGrid` callable."
+                )
+
+            self._callable_object_id = underlying_aggrid_object.id
+
+    def build(self):
+        return dcc.Loading(
+            html.Div(
+                [
+                    html.H3(self.title, className="table-title") if self.title else None,
+                    html.Div(dag.AgGrid(**({"id": self._callable_object_id} if self.actions else {})), id=self.id),
+                ],
```

Comment:
Max, can we fix one bug in this PR (and then I can solve this one [from the comment](https://github.com/mckinsey/vizro/pull/289/files#r1497888111))? The bug happens when no action is associated with the AgGrid/Table object, but this AgGrid/Table model is used as the input of some custom action. The bug says something like: "There is no component with the ID: {ID}, but it's used as dash.Input in the callback..."


All we need to avoid this bug to occur is:
1.  to change this line:
`html.Div(dag.AgGrid(**({"id": self._callable_object_id} if self.actions else {})), id=self.id),`
to
`html.Div(dag.AgGrid(id=self._callable_object_id), id=self.id),`

2. and to remove `if self.actions:` statement completely from the pre_build. 

We should apply this change in vm.Table as well.

Reply:
Ok done - will push in a sec. Can you then check if I understood this correctly?